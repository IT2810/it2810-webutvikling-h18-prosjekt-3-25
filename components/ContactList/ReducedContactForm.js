import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';

export default class ReducedContactForm extends React.Component {
    render() {
        return (
            //TouchableHighlight gjør elementet trykkbart og lager effekten
            //man ser når man trykker på det
            <TouchableHighlight onPress={this.props.onPress}>
                <View style={styles.contactFormContainer}>
                    <Avatar
                        medium
                        rounded
                        title={this.props.name.split(" ").map(x => x[0]).join("")}
                    />
                    <View style={styles.names}>
                        <Text style={styles.text}>{this.props.name}</Text>
                    </View>
                </View>
            </TouchableHighlight>
        );
    }
}

const styles = StyleSheet.create({

    contactFormContainer: {
        flex: 1,
        flexDirection: "row",
        paddingTop: 10,
        paddingBottom: 10,
        borderBottomWidth: 1,
        borderBottomColor: "#aaa",
    },

    names: {
        justifyContent: "center",
    },

    text: {
        paddingLeft: 20,
        fontSize: 18,
    }
});
