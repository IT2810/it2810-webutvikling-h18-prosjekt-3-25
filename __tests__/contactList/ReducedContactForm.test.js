import React from 'react';
import ReducedContactForm from '../../components/ContactList/ReducedContactForm.js';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

it("renders correctly", () => {
    const tree = renderer.create(<ReducedContactForm name="Test Person" />).toJSON();
    expect(tree).toMatchSnapshot();
})
