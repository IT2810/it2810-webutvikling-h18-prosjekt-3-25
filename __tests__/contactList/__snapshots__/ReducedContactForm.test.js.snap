// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 50,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 50,
          },
          Object {
            "borderRadius": 25,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 25,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 25,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          TP
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 20,
          }
        }
      >
        Test Person
      </Text>
    </View>
  </View>
</View>
`;
